### ISSUE TITLE
.a archive stripping corrupts thin archives

### ISSUE THREAD

**Issue by okhlybov** [Alexpux/MSYS2-packages#1175]
Makepkg's 'strip' option corrupts 'thin' archives created by `ar cT` command leaving out just a `!<thin>` header in the resulting file.

**Comment by Alexpux** [comment]
Here is strip implementation for MSYS2 pacman:
https://github.com/Alexpux/MSYS2-packages/blob/master/pacman/0000-pacman-msysize.patch#L492

Look at it maybe you will see some thing incorrect. I will look too

**Comment by okhlybov** [comment]
My guess is that the solution should be straightforward: skip thin archives.
The point is that they do not contain object files by themselves (kind of import libraries for static linking) but only the references to other archives therefore there is nothing to strip.
Such archives can be easily identified by the !<thin> header instead of normal !<arch>.

**Comment by Alexpux** [comment]
@okhlybov can you provide some such archive for testing?

**Comment by okhlybov** [comment]
Sure. Here `a.a` is a normal archive and `thin.a` is a thin archive referring to it.
[thin.zip](https://github.com/Alexpux/MSYS2-packages/files/1743993/thin.zip)

**Comment by Alexpux** [comment]
Are you sure that thin archive is stripped? I don't see how it can be in logic.
See ````/usr/share/makepkg/tidy/strip.sh````
Running ````file -bi thin.a```` return ````application/octet-stream; charset=binary```` and it going in case to continue without stripping

**Comment by okhlybov** [comment]
Indeed, thin archives don't get stripped, so the problem is not that trivial. I guess there might be a subtle bug in binutils.
 Here is a real archive which manifests the anomaly: [thin.zip](https://github.com/Alexpux/MSYS2-packages/files/1745509/thin.zip)  In short, if a thin archive is created from an unstripped normal archive which is gets stripped afterwards, the thin one *might* get out of sync and thus become invalid even though the archive it refers to looks intact.

**Comment by Alexpux** [comment]
@okhlybov if you think that this is bug in binutils then address it to binutils developers.
